azext_edge/edge/providers/rpsaas/adr/base.py

```diff
@@ -0,0 +1,210 @@
+# coding=utf-8
+# ----------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License file in the project root for license information.
+# ----------------------------------------------------------------------------------------------
+
+from typing import Any, Dict, Optional
+from knack.log import get_logger
+from azure.cli.core.azclierror import (
+    ResourceNotFoundError,
+    RequiredArgumentMissingError,
+    ValidationError
+)
+
+from .user_strings import (
+    CUSTOM_LOCATION_DOES_NOT_EXIST_ERROR,
+    CUSTOM_LOCATION_NOT_FOUND_ERROR,
+    CLUSTER_NOT_FOUND_ERROR,
+    CLUSTER_OFFLINE,
+    MISSING_CLUSTER_CUSTOM_LOCATION_ERROR,
+    MISSING_EXTENSION_ERROR,
+    MULTIPLE_CUSTOM_LOCATIONS_ERROR,
+    MULTIPLE_POSSIBLE_ITEMS_ERROR
+)
+from ..base_provider import RPSaaSBaseProvider
+from ....util import build_query
+from ....common import ResourceTypeMapping
+
+logger = get_logger(__name__)
+API_VERSION = "2023-11-01-preview"
+
+
+# @vilit - prob generalize this even more
+class ADRBaseProvider(RPSaaSBaseProvider):
+    def __init__(
+        self, cmd, resource_type: str
+    ):
+        super(ADRBaseProvider, self).__init__(
+            cmd=cmd,
+            api_version=API_VERSION,
+            resource_type=resource_type,
+            required_extension="microsoft.deviceregistry.assets"
+        )
+
+    def delete(
```

Comment:
There is a delete here as well as the RPSaaSBaseProvider?

Reply:
This delete has a show and check and is meant to override the parent delete - if the cluster is offline and you send a delete to ADR, wonky things happen :D... do you want me to just rename this to check_and_delete to differentiate between the two? the ADR methods do not use the parent delete

Reply:
What if we just had one delete residing in the most base class and that had a parameter like `check_cluster_connectivity: bool`